Library/OcCpuLib/OcCpuLib.c

```diff
@@ -1201,7 +1201,11 @@ ScanAmdProcessor (
         CofVid           = AsmReadMsr64 (K10_PSTATE_STATUS);
         CoreFrequencyID  = BitFieldRead64 (CofVid, 0, 7);
         CoreDivisorID    = BitFieldRead64 (CofVid, 8, 13);
-        Cpu->MaxBusRatio = (UINT8) (CoreFrequencyID / CoreDivisorID * 2);
+        if (CoreDivisorID > 0ULL){
```

Comment:
This needs documentation. I have a Ryzen machine myself, can confirm this bug and, locally, have worked around this in pretty much the same way, however I have not bothered (yet) to research a proper solution to the problem. At very least this needs a comment mentioning the HV situation, but I'm not sure whether we want to allow this condition on bare metal at all.

Reply:
I am fine with just a comment mentioning that `CoreDivisorID` may be 0 on Hypervisor and it needs further research for this exact pull request. Here and below.